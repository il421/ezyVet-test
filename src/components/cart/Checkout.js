import React from 'react';

import myPhoto from '../../assets/img/my-photo.png';
import '../../styles/cart/checkout.css';
import '../../styles/common/button.css';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';


const Checkout = ({ handleDisplayCheckoutModal }) => {
  return (
    <div className="checkout">
      <button className="checkout__close" onClick={ () => handleDisplayCheckoutModal() }>
        <FontAwesomeIcon icon="times" size="2x" />
      </button>

      <div className="checkout__wrapper">
        <div className="checkout__desc">
          <h2>Suglobov Ilya</h2>
          <div>I used:</div>
          <ul>
            <li>React, Create React App (I can also use Webpack to build, and optimize projects)</li>
            <li>Redux, Redux-Persist, Redux-Thunk</li>
            <li>HTML5, CSS3 (usually I use SCSS, for the project with simple styles it is not necessary)</li>
            <li>Jest, Enzyme</li>
          </ul>
          <div>Thank you!</div>
        </div>
      </div>
    </div>
  );
};

export default Checkout;
