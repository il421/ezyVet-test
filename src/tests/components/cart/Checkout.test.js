import React from 'react';
import { shallow } from 'enzyme';
import Checkout from '../../../components/cart/Checkout.js';

let wrapper, handleDisplayCheckoutModal;

beforeEach(() => {
  handleDisplayCheckoutModal = jest.fn();

  wrapper = shallow(
    <Checkout
      handleDisplayCheckoutModal={ handleDisplayCheckoutModal }
    />
  );
});

test('should render Checkout without crashing', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call handleDisplayCheckoutModal if click checkout close btn', () => {
  wrapper.find('.checkout__close').simulate('click');
  expect(handleDisplayCheckoutModal).toHaveBeenCalled();
});
