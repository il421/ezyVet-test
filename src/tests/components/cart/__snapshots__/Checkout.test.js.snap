// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Checkout without crashing 1`] = `
<div
  className="checkout"
>
  <button
    className="checkout__close"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="times"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
  <div
    className="checkout__wrapper"
  >
    <div
      className="checkout__desc"
    >
      <h2>
        Suglobov Ilya
      </h2>
      <div>
        I used:
      </div>
      <ul>
        <li>
          React, Create React App (I can also use Webpack to build, and optimize projects)
        </li>
        <li>
          Redux, Redux-Persist, Redux-Thunk
        </li>
        <li>
          HTML5, CSS3 (usually I use SCSS, for the project with simple styles it is not necessary)
        </li>
        <li>
          Jest, Enzyme
        </li>
      </ul>
      <div>
        Thank you!
      </div>
    </div>
  </div>
</div>
`;
